biodatasets/ehr_rel/ehr_rel.py

```diff
@@ -0,0 +1,195 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+EHR-Rel is a novel open-source1 biomedical concept relatedness dataset consisting of 3630 concept pairs, six times more
+ than the largest existing dataset.  Instead of manually selecting and pairing concepts as done in previous work,
+ the dataset is sampled from EHRs to ensure concepts are relevant for the EHR concept retrieval task.
+ A detailed analysis of the concepts in the dataset reveals a far larger coverage compared to existing datasets.
+"""
+
+import csv
+from pathlib import Path
+from typing import Dict, Iterator, List, Tuple
+
+import datasets
+
+from utils import schemas
+from utils.configs import BigBioConfig
+from utils.constants import Tasks
+
+_CITATION = """\
+@inproceedings{schulz-etal-2020-biomedical,
+    title = {Biomedical Concept Relatedness {--} A large {EHR}-based benchmark},
+    author = {Schulz, Claudia  and
+      Levy-Kramer, Josh  and
+      Van Assel, Camille  and
+      Kepes, Miklos  and
+      Hammerla, Nils},
+    booktitle = {Proceedings of the 28th International Conference on Computational Linguistics},
+    month = {dec},
+    year = {2020},
+    address = {Barcelona, Spain (Online)},
+    publisher = {International Committee on Computational Linguistics},
+    url = {https://aclanthology.org/2020.coling-main.577},
+    doi = {10.18653/v1/2020.coling-main.577},
+    pages = {6565--6575},
+    }
+"""
+
+_DATASETNAME = "ehr_rel"
+
+_DESCRIPTION = """\
+EHR-Rel is a novel open-source1 biomedical concept relatedness dataset consisting of 3630 concept pairs, six times more
+than the largest existing dataset.  Instead of manually selecting and pairing concepts as done in previous work,
+the dataset is sampled from EHRs to ensure concepts are relevant for the EHR concept retrieval task.
+A detailed analysis of the concepts in the dataset reveals a far larger coverage compared to existing datasets.
+"""
+
+_HOMEPAGE = "https://aclanthology.org/2020.coling-main.577/"
+
+_LICENSE = "Apache License 2.0"
+
+_URLS = {
+    _DATASETNAME: "https://github.com/babylonhealth/EHR-Rel/archive/refs/heads/master.zip",
+}
+
+_SUPPORTED_TASKS = [Tasks.SEMANTIC_SIMILARITY]
+
+
+_SOURCE_VERSION = "1.0.0"
+
+_BIGBIO_VERSION = "1.0.0"
+
+
+class EHRRelDataset(datasets.GeneratorBasedBuilder):
+    """Dataset for EHR-Rel Corpus"""
+
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    BIGBIO_VERSION = datasets.Version(_BIGBIO_VERSION)
+
+    BUILDER_CONFIGS = [
+        BigBioConfig(
+            name="ehr_rel_source",
+            version=SOURCE_VERSION,
+            description="EHR-Rel source schema",
+            schema="source",
+            subset_id="ehr_rel",
+        ),
+        BigBioConfig(
```

Comment:
The dataset seems to have 2 `subset_id` (EHR-RelA and EHR-RelB) and it seems that EHR-RelA is used as `negative examples`. For the `source` schema could you please mantain this division? This way we preserve the data as was generated

Reply:
Done.